examples/go.mod

```diff
@@ -4,10 +4,18 @@ go 1.20
 
 require (
 	github.com/mattn/go-sqlite3 v1.14.22
+	github.com/segmentio/kafka-go v0.4.47
```

Comment:
do we want to wait until https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/650, which commits each example's individual `go.mod` files and then update this PR to follow that?

Reply:
Sure doesn't look like a big rebase whether this or the other one gets merged first.